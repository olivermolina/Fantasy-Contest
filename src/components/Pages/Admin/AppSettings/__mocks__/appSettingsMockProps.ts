export const appSettingsMockProps = {
  appSettings: {
    BONUS_CREDIT_FREE_ENTRY_EQUIVALENT: '25',
    STAKE_TYPE_FREE_ENTRY: 'ALL_IN,INSURED',
    NUMBER_OF_PLAYERS_FREE_ENTRY: '2,3,4',
    MAX_MATCH_DEPOSIT_AMOUNT: '100',
    DEPOSIT_AMOUNT_OPTIONS: '10,25,50,75,100,200',
    RELOAD_BONUS_TYPE: 'FLAT',
    RELOAD_BONUS_AMOUNT: '0',
    MAX_RETENTION_BONUS: '50',
    RETENTION_BONUS_WEEKLY_CHANCE: '1',
    RETENTION_BONUS_MATCH_MULTIPLIER: '1',
    REFERRAL_CREDIT_AMOUNT: '25',
    REFERRAL_CUSTOM_TEXT:
      'Refer a friend from the link or code in your profile and get $25 in bonus credit!!',
    MIN_BET_AMOUNT: '1',
    MAX_BET_AMOUNT: '50',
    MAX_MARKET_ODDS: '150',
    MIN_MARKET_ODDS: '-150',
    initial: false,
    CHALLENGE_PROMO_MESSAGE:
      'Pick 2-4 players. Predict if they will get MORE or LESS than their projection.',
    REPEAT_ENTRIES_LIMIT: '1',
  },
  onSubmit: () => console.log('onSubmit'),
};
