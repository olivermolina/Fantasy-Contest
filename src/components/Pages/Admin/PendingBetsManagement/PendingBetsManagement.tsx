import React, { useState } from 'react';
import { DataGrid, GridColDef, GridRowsProp } from '@mui/x-data-grid';
import { Button, Stack } from '@mui/material';
import PendingPickDialog from './PendingPickDialog';
import { BetStatus } from '@prisma/client';

type Props = {
  settlePick(currentRow: RowModel, betStatus: BetStatus): unknown;
  data: readonly RowModel[];
  updateBetLeg: (leg: LegRowModel, status: BetStatus) => void;
};

export interface LegRowModel {
  id: string;
  name: string;
  type: string;
  odds: number;
  category: string;
  total: number;
  total_stat: number | null;
  status: string;
}

export interface RowModel {
  ticket: string;
  placed: Date;
  userPhone: string;
  status: string;
  name: string;
  username: string;
  description: string;
  riskWin: string;
  type: string;
  legs: LegRowModel[];
}

export const PendingBetsManagement = (props: Props) => {
  const rows: GridRowsProp<RowModel> = props.data;
  const [selectedRow, setSelectedRow] = useState<RowModel | undefined>();
  const [betStatus, setBetStatus] = useState<BetStatus | undefined>();
  const [open, setOpen] = useState(false);

  const columns: GridColDef[] = [
    { flex: 1, field: 'ticket', headerName: 'Ticket #' },
    {
      flex: 1,
      field: 'placed',
      headerName: 'Placed',
      valueFormatter: (params) => params.value.toLocaleString(),
    },
    { flex: 1, field: 'userPhone', headerName: 'User/Phone' },
    { flex: 1, field: 'username', headerName: 'Username' },
    {
      flex: 1,
      field: 'description',
      headerName: 'Description',
    },
    { flex: 1, field: 'riskWin', headerName: 'Risk/Win' },
    {
      flex: 1,
      field: 'view',
      headerName: 'Legs',
      renderCell: (params) => {
        const onClick = () => {
          const currentRow = params.row as (typeof rows)[0];
          setSelectedRow(currentRow);
          setBetStatus(currentRow.status as BetStatus);
          setOpen(true);
        };

        return (
          <Stack direction="row" spacing={2}>
            <Button
              color="primary"
              size="small"
              onClick={onClick}
              variant="outlined"
            >
              View
            </Button>
          </Stack>
        );
      },
    },
    {
      flex: 2,
      field: 'action',
      headerName: 'Actions',
      renderCell: (params) => {
        const onClick = (betStatus: BetStatus) => {
          const currentRow = params.row as (typeof rows)[0];
          setSelectedRow(currentRow);
          setOpen(true);
          setBetStatus(betStatus);
        };

        return (
          <div className={'flex gap-1'}>
            <Button
              color="warning"
              size="small"
              onClick={() => onClick(BetStatus.WIN)}
              variant={'contained'}
            >
              WIN
            </Button>
            <Button
              color="warning"
              size="small"
              onClick={() => onClick(BetStatus.LOSS)}
              variant={'contained'}
            >
              LOSS
            </Button>
            <Button
              color="warning"
              size="small"
              onClick={() => onClick(BetStatus.REFUNDED)}
              variant={'contained'}
            >
              Refund
            </Button>
          </div>
        );
      },
    },
  ];

  const clearSelectedRow = () => {
    setSelectedRow(undefined);
    handleClose();
  };
  const handleClose = () => {
    setOpen(false);
  };

  const setSelectedBetStatus = (betStatus: BetStatus) => {
    setBetStatus(betStatus);
  };

  return (
    <div className={'h-[75vh] lg:h-[80vh] w-full'}>
      <DataGrid getRowId={(row) => row.ticket} rows={rows} columns={columns} />
      <PendingPickDialog
        settlePick={props.settlePick}
        clearSelectedRow={clearSelectedRow}
        open={open}
        row={selectedRow}
        handleClose={handleClose}
        betStatus={betStatus}
        setSelectedBetStatus={setSelectedBetStatus}
        updateBetLeg={props.updateBetLeg}
      />
    </div>
  );
};
