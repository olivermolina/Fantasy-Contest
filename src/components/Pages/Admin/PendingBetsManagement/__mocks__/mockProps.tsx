export const mockProps = {
  onClickDeleteRow: () => {
    console.log('click');
  },
  data: [
    {
      ticket: '1',
      placed: new Date('1970-01-01T01:00:00.000Z'),
      userPhone: '1',
      username: 'username',
      description: '1',
      riskWin: '1',
      legs: [
        {
          id: '1',
          name: 'name',
          type: 'type',
          odds: 100,
          category: 'Assist',
          total: 5,
        },
      ],
    },
    {
      ticket: '2',
      placed: new Date('1970-01-01T01:00:00.000Z'),
      userPhone: '2',
      username: 'username',
      description: '2',
      riskWin: '2',
      legs: [
        {
          id: '2',
          name: 'name',
          type: 'type',
          odds: 100,
          category: 'Assist',
          total: 5,
        },
      ],
    },
    {
      ticket: '3',
      placed: new Date('1970-01-01T01:00:00.000Z'),
      userPhone: '3',
      username: 'username',
      description: '3',
      riskWin: '3',
      legs: [
        {
          id: '3',
          name: 'name',
          type: 'type',
          odds: 100,
          category: 'Assist',
          total: 5,
        },
      ],
    },
  ],
};
