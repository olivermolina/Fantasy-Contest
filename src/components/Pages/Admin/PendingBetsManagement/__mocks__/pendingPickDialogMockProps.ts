export const pendingPickDialogProps = {
  handleClose: () => {
    console.log('handleClose');
  },
  open: false,
  settlePick: () => {
    console.log('settlePick');
  },
  row: undefined,
  clearSelectedRow: () => {
    console.log('clearSelectedRow');
  },
  setSelectedBetStatus: () => {
    console.log('setSelectedBetStatus');
  },
  updateBetLeg: () => {
    console.log('updateBetLeg');
  },
  isViewOnly: false,
};
