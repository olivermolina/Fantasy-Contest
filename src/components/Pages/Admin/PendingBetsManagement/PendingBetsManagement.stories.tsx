import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import { PendingBetsManagement } from './PendingBetsManagement';
import { mockProps } from './__mocks__/mockProps';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  component: PendingBetsManagement,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {
    backgroundColor: { control: 'color' },
  },
} as ComponentMeta<typeof PendingBetsManagement>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof PendingBetsManagement> = (args) => (
  <PendingBetsManagement {...args} />
);

export const Primary = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
Primary.args = mockProps;
