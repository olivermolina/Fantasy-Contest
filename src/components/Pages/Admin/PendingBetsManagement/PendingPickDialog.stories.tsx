import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';

import PendingPickDialog from './PendingPickDialog';
import { pendingPickDialogProps } from '~/components/Pages/Admin/PendingBetsManagement/__mocks__/pendingPickDialogMockProps';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  component: PendingPickDialog,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {
    backgroundColor: { control: 'color' },
  },
} as ComponentMeta<typeof PendingPickDialog>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof PendingPickDialog> = (args) => (
  <PendingPickDialog {...args} />
);

export const Dialog = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
Dialog.args = pendingPickDialogProps;
