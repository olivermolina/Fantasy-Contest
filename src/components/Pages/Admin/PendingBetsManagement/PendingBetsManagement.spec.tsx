import React from 'react';
import { render } from '@testing-library/react';
import { PendingBetsManagement } from './PendingBetsManagement';
import { mockProps } from './__mocks__/mockProps';

describe('PendingBetsManagement', () => {
  it('should render', () => {
    const { container } = render(<PendingBetsManagement {...mockProps} />);
    expect(container).toMatchSnapshot();
  });
});
