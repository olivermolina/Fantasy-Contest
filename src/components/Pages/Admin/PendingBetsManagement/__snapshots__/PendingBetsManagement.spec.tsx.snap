// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PendingBetsManagement should render 1`] = `
<div>
  <div
    class="h-[75vh] lg:h-[80vh] w-full"
  >
    <div
      aria-colcount="8"
      aria-multiselectable="false"
      aria-rowcount="4"
      class="MuiDataGrid-root MuiDataGrid-root--densityStandard MuiDataGrid-withBorderColor css-1lymaxv-MuiDataGrid-root"
      role="grid"
    >
      <div
        class="MuiDataGrid-main css-204u17-MuiDataGrid-main"
      >
        <div
          class="MuiDataGrid-columnHeaders MuiDataGrid-withBorderColor css-1iyq7zh-MuiDataGrid-columnHeaders"
          style="min-height: 56px; max-height: 56px; line-height: 56px;"
        >
          <div
            class="MuiDataGrid-columnHeadersInner css-gl260s-MuiDataGrid-columnHeadersInner"
            role="rowgroup"
            style="transform: translate3d(0px, 0px, 0px);"
          >
            <div
              aria-rowindex="1"
              class="css-yrdy0g-MuiDataGrid-columnHeaderRow"
              role="row"
            >
              <div
                aria-colindex="1"
                aria-label="Ticket #"
                aria-sort="none"
                class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable MuiDataGrid-withBorderColor"
                data-field="ticket"
                role="columnheader"
                style="height: 56px; width: 0px; min-width: 0; max-width: 0;"
                tabindex="0"
              >
                <div
                  class="MuiDataGrid-columnHeaderDraggableContainer"
                  draggable="false"
                >
                  <div
                    class="MuiDataGrid-columnHeaderTitleContainer"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderTitleContainerContent"
                    >
                      <div
                        aria-label=""
                        class="MuiDataGrid-columnHeaderTitle css-t89xny-MuiDataGrid-columnHeaderTitle"
                        data-mui-internal-clone-element="true"
                      >
                        Ticket #
                      </div>
                    </div>
                    <div
                      class="MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer"
                    >
                      <button
                        aria-label="Sort"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                        tabindex="-1"
                        title="Sort"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-ptiqhd-MuiSvgIcon-root"
                          data-testid="ArrowUpwardIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                  <div
                    class="MuiDataGrid-menuIcon"
                  >
                    <button
                      aria-expanded="false"
                      aria-haspopup="menu"
                      aria-label="Menu"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiDataGrid-menuIconButton css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                      data-mui-internal-clone-element="true"
                      id=":r4:"
                      tabindex="-1"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                        data-testid="TripleDotsVerticalIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
                <div
                  class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                  style="min-height: 56px; opacity: 1;"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                    data-testid="SeparatorIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11 19V5h2v14z"
                    />
                  </svg>
                </div>
              </div>
              <div
                aria-colindex="2"
                aria-label="Placed"
                aria-sort="none"
                class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable MuiDataGrid-withBorderColor"
                data-field="placed"
                role="columnheader"
                style="height: 56px; width: 0px; min-width: 0; max-width: 0;"
                tabindex="-1"
              >
                <div
                  class="MuiDataGrid-columnHeaderDraggableContainer"
                  draggable="false"
                >
                  <div
                    class="MuiDataGrid-columnHeaderTitleContainer"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderTitleContainerContent"
                    >
                      <div
                        aria-label=""
                        class="MuiDataGrid-columnHeaderTitle css-t89xny-MuiDataGrid-columnHeaderTitle"
                        data-mui-internal-clone-element="true"
                      >
                        Placed
                      </div>
                    </div>
                    <div
                      class="MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer"
                    >
                      <button
                        aria-label="Sort"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                        tabindex="-1"
                        title="Sort"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-ptiqhd-MuiSvgIcon-root"
                          data-testid="ArrowUpwardIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                  <div
                    class="MuiDataGrid-menuIcon"
                  >
                    <button
                      aria-expanded="false"
                      aria-haspopup="menu"
                      aria-label="Menu"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiDataGrid-menuIconButton css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                      data-mui-internal-clone-element="true"
                      id=":ra:"
                      tabindex="-1"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                        data-testid="TripleDotsVerticalIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
                <div
                  class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                  style="min-height: 56px; opacity: 1;"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                    data-testid="SeparatorIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11 19V5h2v14z"
                    />
                  </svg>
                </div>
              </div>
              <div
                aria-colindex="3"
                aria-label="User/Phone"
                aria-sort="none"
                class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable MuiDataGrid-withBorderColor"
                data-field="userPhone"
                role="columnheader"
                style="height: 56px; width: 0px; min-width: 0; max-width: 0;"
                tabindex="-1"
              >
                <div
                  class="MuiDataGrid-columnHeaderDraggableContainer"
                  draggable="false"
                >
                  <div
                    class="MuiDataGrid-columnHeaderTitleContainer"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderTitleContainerContent"
                    >
                      <div
                        aria-label=""
                        class="MuiDataGrid-columnHeaderTitle css-t89xny-MuiDataGrid-columnHeaderTitle"
                        data-mui-internal-clone-element="true"
                      >
                        User/Phone
                      </div>
                    </div>
                    <div
                      class="MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer"
                    >
                      <button
                        aria-label="Sort"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                        tabindex="-1"
                        title="Sort"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-ptiqhd-MuiSvgIcon-root"
                          data-testid="ArrowUpwardIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                  <div
                    class="MuiDataGrid-menuIcon"
                  >
                    <button
                      aria-expanded="false"
                      aria-haspopup="menu"
                      aria-label="Menu"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiDataGrid-menuIconButton css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                      data-mui-internal-clone-element="true"
                      id=":rg:"
                      tabindex="-1"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                        data-testid="TripleDotsVerticalIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
                <div
                  class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                  style="min-height: 56px; opacity: 1;"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                    data-testid="SeparatorIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11 19V5h2v14z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiDataGrid-virtualScroller css-axafay-MuiDataGrid-virtualScroller"
          style="overflow-x: hidden;"
        >
          <div
            class="MuiDataGrid-virtualScrollerContent MuiDataGrid-virtualScrollerContent--overflowed css-1kwdphh-MuiDataGrid-virtualScrollerContent"
            style="width: auto; height: 156px; min-height: auto;"
          >
            <div
              class="MuiDataGrid-virtualScrollerRenderZone css-s1v7zr-MuiDataGrid-virtualScrollerRenderZone"
              style="transform: translate3d(0px, 0px, 0px);"
            >
              <div
                aria-rowindex="2"
                aria-selected="false"
                class="MuiDataGrid-row"
                data-id="1"
                data-rowindex="0"
                role="row"
                style="max-height: 52px; min-height: 52px;"
              >
                <div
                  aria-colindex="1"
                  aria-colspan="1"
                  class="MuiDataGrid-cell MuiDataGrid-cell--textLeft MuiDataGrid-withBorderColor"
                  data-colindex="0"
                  data-field="ticket"
                  role="cell"
                  style="min-width: 0; max-width: 0; min-height: 52px; max-height: 52px;"
                  tabindex="-1"
                >
                  <div
                    class="MuiDataGrid-cellContent"
                    title="1"
                  >
                    1
                  </div>
                </div>
                <div
                  aria-colindex="2"
                  aria-colspan="1"
                  class="MuiDataGrid-cell MuiDataGrid-cell--textLeft MuiDataGrid-withBorderColor"
                  data-colindex="1"
                  data-field="placed"
                  role="cell"
                  style="min-width: 0; max-width: 0; min-height: 52px; max-height: 52px;"
                  tabindex="-1"
                >
                  <div
                    class="MuiDataGrid-cellContent"
                    title="1/1/1970, 1:00:00 AM"
                  >
                    1/1/1970, 1:00:00 AM
                  </div>
                </div>
                <div
                  aria-colindex="3"
                  aria-colspan="1"
                  class="MuiDataGrid-cell MuiDataGrid-cell--textLeft MuiDataGrid-withBorderColor"
                  data-colindex="2"
                  data-field="userPhone"
                  role="cell"
                  style="min-width: 0; max-width: 0; min-height: 52px; max-height: 52px;"
                  tabindex="-1"
                >
                  <div
                    class="MuiDataGrid-cellContent"
                    title="1"
                  >
                    1
                  </div>
                </div>
              </div>
              <div
                aria-rowindex="3"
                aria-selected="false"
                class="MuiDataGrid-row"
                data-id="2"
                data-rowindex="1"
                role="row"
                style="max-height: 52px; min-height: 52px;"
              >
                <div
                  aria-colindex="1"
                  aria-colspan="1"
                  class="MuiDataGrid-cell MuiDataGrid-cell--textLeft MuiDataGrid-withBorderColor"
                  data-colindex="0"
                  data-field="ticket"
                  role="cell"
                  style="min-width: 0; max-width: 0; min-height: 52px; max-height: 52px;"
                  tabindex="-1"
                >
                  <div
                    class="MuiDataGrid-cellContent"
                    title="2"
                  >
                    2
                  </div>
                </div>
                <div
                  aria-colindex="2"
                  aria-colspan="1"
                  class="MuiDataGrid-cell MuiDataGrid-cell--textLeft MuiDataGrid-withBorderColor"
                  data-colindex="1"
                  data-field="placed"
                  role="cell"
                  style="min-width: 0; max-width: 0; min-height: 52px; max-height: 52px;"
                  tabindex="-1"
                >
                  <div
                    class="MuiDataGrid-cellContent"
                    title="1/1/1970, 1:00:00 AM"
                  >
                    1/1/1970, 1:00:00 AM
                  </div>
                </div>
                <div
                  aria-colindex="3"
                  aria-colspan="1"
                  class="MuiDataGrid-cell MuiDataGrid-cell--textLeft MuiDataGrid-withBorderColor"
                  data-colindex="2"
                  data-field="userPhone"
                  role="cell"
                  style="min-width: 0; max-width: 0; min-height: 52px; max-height: 52px;"
                  tabindex="-1"
                >
                  <div
                    class="MuiDataGrid-cellContent"
                    title="2"
                  >
                    2
                  </div>
                </div>
              </div>
              <div
                aria-rowindex="4"
                aria-selected="false"
                class="MuiDataGrid-row MuiDataGrid-row--lastVisible"
                data-id="3"
                data-rowindex="2"
                role="row"
                style="max-height: 52px; min-height: 52px;"
              >
                <div
                  aria-colindex="1"
                  aria-colspan="1"
                  class="MuiDataGrid-cell MuiDataGrid-cell--textLeft MuiDataGrid-withBorderColor"
                  data-colindex="0"
                  data-field="ticket"
                  role="cell"
                  style="min-width: 0; max-width: 0; min-height: 52px; max-height: 52px;"
                  tabindex="-1"
                >
                  <div
                    class="MuiDataGrid-cellContent"
                    title="3"
                  >
                    3
                  </div>
                </div>
                <div
                  aria-colindex="2"
                  aria-colspan="1"
                  class="MuiDataGrid-cell MuiDataGrid-cell--textLeft MuiDataGrid-withBorderColor"
                  data-colindex="1"
                  data-field="placed"
                  role="cell"
                  style="min-width: 0; max-width: 0; min-height: 52px; max-height: 52px;"
                  tabindex="-1"
                >
                  <div
                    class="MuiDataGrid-cellContent"
                    title="1/1/1970, 1:00:00 AM"
                  >
                    1/1/1970, 1:00:00 AM
                  </div>
                </div>
                <div
                  aria-colindex="3"
                  aria-colspan="1"
                  class="MuiDataGrid-cell MuiDataGrid-cell--textLeft MuiDataGrid-withBorderColor"
                  data-colindex="2"
                  data-field="userPhone"
                  role="cell"
                  style="min-width: 0; max-width: 0; min-height: 52px; max-height: 52px;"
                  tabindex="-1"
                >
                  <div
                    class="MuiDataGrid-cellContent"
                    title="3"
                  >
                    3
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiDataGrid-footerContainer MuiDataGrid-withBorderColor css-wop1k0-MuiDataGrid-footerContainer"
      >
        <div />
        <div
          class="MuiTablePagination-root css-rtrcn9-MuiTablePagination-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
          >
            <div
              class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
            />
            <p
              class="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
              id=":r2:"
            >
              Rows per page:
            </p>
            <div
              class="MuiInputBase-root MuiInputBase-colorPrimary css-16c50h-MuiInputBase-root-MuiTablePagination-select"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby=":r2: :r1:"
                class="MuiSelect-select MuiTablePagination-select MuiSelect-standard MuiInputBase-input css-194a1fa-MuiSelect-select-MuiInputBase-input"
                id=":r1:"
                role="button"
                tabindex="0"
              >
                100
              </div>
              <input
                aria-hidden="true"
                aria-invalid="false"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                tabindex="-1"
                value="100"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiTablePagination-selectIcon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
            <p
              class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
            >
              1–3 of 3
            </p>
            <div
              class="MuiTablePagination-actions"
            >
              <button
                aria-label="Go to previous page"
                class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                disabled=""
                tabindex="-1"
                title="Go to previous page"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="KeyboardArrowLeftIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                  />
                </svg>
              </button>
              <button
                aria-label="Go to next page"
                class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                disabled=""
                tabindex="-1"
                title="Go to next page"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="KeyboardArrowRightIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
